src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -102,9 +111,22 @@ public Task RunAsync(ActionRunStage stage)
             actionRunner2.Action = cleanOutputsStep;
             actionRunner2.Stage = ActionRunStage.Main;
             actionRunner2.Condition = "always()";
-            ExecutionContext.RegisterNestedStep(actionRunner2, inputsData, location, Environment, true);
+            var cleanUpStep = ExecutionContext.RegisterNestedStep(actionRunner2, inputsData, location, Environment, true);
```

Comment:
You don't need a step for setting outputs, inline the logic here.
the handler might look like:
- gather steps from composite action.
- run steps with compositeStepsRunner.
- set outputs.

Reply:
Good idea - thanks!